src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -398,6 +398,64 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     unreached();
             }
         }
+        else if (isRMW)
+        {
+            assert(!hasImmediateOperand);
+            assert(!HWIntrinsicInfo::SupportsContainment(intrin.id));
+
+            // Move the RMW register out of the way and do not pass it to the emit.
+
+            if (HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrin.id))
+            {
+                // op1Reg contains a mask, op2Reg contains the RMW register.
+
+                if (targetReg != op2Reg)
+                {
+                    assert(targetReg != op3Reg);
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op2Reg, /* canSkip */ true);
```

Comment:
It's the same general `mov` needed for any `RMW` instruction since you need the destination register and one of the input registers to be the same. The only really different thing is that `movprfx` is specifically the one you want for SVE instructions that have an embedded mask. -- The potentially confusing part here is that it's checking `IsEmbeddedMaskedOperation` when that check as currently implemented just means "I have a mask, but its all true" (I gave feedback that should be renamed above)

So I'm not sure there's really any rules to document here, we certainly don't document it at all for x86 or x64 or any of the existing Arm64 RMW cases.